mypy/typevars.py

```diff
@@ -22,4 +21,4 @@ def fill_typevars(typ: TypeInfo) -> Union[Instance, TupleType]:
 
 
 def has_no_typevars(typ: Type) -> bool:
-    return is_same_type(typ, erase_typevars(typ))
+    return typ == erase_typevars(typ)
```

Comment:
Add a comment explaining what is going on here, since doing equality on types is usually the wrong thing, since it isn't overloaded. Might make more sense for it to be `is`?

Reply:
(on the grounds that explicit is better than implicit, so if we are just doing object identity might as well be obvious about it)

Reply:
Oh lurr I actually didn't realize that `Type` subclasses all had overloaded `__eq__`